modules/meta/arrow-meta-prototype/testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt

```diff
@@ -0,0 +1,65 @@
+package arrow.meta.plugin.testing
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import org.assertj.core.api.Assertions.assertThat
+import java.io.File
+import java.nio.file.Paths
+import io.github.classgraph.ClassGraph
+
+enum class CompilationResult {
+  OK,
+  INTERNAL_ERROR,
+  COMPILATION_ERROR,
+  SCRIPT_EXECUTION_ERROR
+}
+
+data class CompilationData(
+  val sourceFileName: String,
+  val sourceContent: String,
+  val generatedFileContent: String,
+  val generatedClasses: ArrayList<String>,
+  val compilationResult: CompilationResult
+)
+
+fun testCompilation(compilationData: CompilationData) {
+  val kotlinSource = SourceFile.kotlin(compilationData.sourceFileName, compilationData.sourceContent)
+
+  val result = KotlinCompilation().apply {
+    sources = listOf(kotlinSource)
+    classpaths = listOf(
+      // TODO
+      // classpathOf("arrow-annotations:x.x.x")
```

Comment:
if we activate the mode in which the plugin includes what's on the test classpath we can just add these dependencies to the build file itself

Reply:
Ok! I'm waiting for the new release of Arrow, because I didn't find `higherkind`in the current release. When it's ready, I was thinking about:

* `build.gradle`
```
dependencies {
    testRuntime "io.arrow-kt:arrow-annotations:0.10.1"
    ...
```
* `Compilation.kt`
```
classpaths = listOf(classpathOf("arrow-annotations:0.10.1"))
```